ide/src/extension.ts

```diff
@@ -1,13 +1,16 @@
 import * as vscode from "vscode";
 import _ from "lodash";
-import { CallFlowistry, log, show_error } from "./vsc_utils";
+import { FocusStatusBarState, render_status_bar } from "./focus_utils";
+import { CallFlowistry, last_error, log, show_error } from "./vsc_utils";
 
 import { decompose } from "./decompose";
 import { focus, focus_mark, focus_unmark, focus_select } from "./focus";
 import { setup } from "./setup";
 
 import "./app.scss";
 
+export let focus_status_bar_item: vscode.StatusBarItem;
```

Comment:
Mutable global variables feels really messy - brainstorming better ways to manage state.

Reply:
Determination after conversation was to use a class to encapsulate global state.

Reply:
Not resolving yet since this can't be encapsulated in `FocusMode` only. Same with `tdcp` also defined in `extension.ts`.